Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs

```diff
@@ -1674,5 +1676,85 @@ public async Task TestCommandWithCommentEmbeddedAsync()
                 Assert.AreEqual("--", reader.GetString(0));
             }
         }
+
+        [Test]
+        public void TestExecuteNonQueryReturnsCorrectRowCountForUploadWithMultipleFiles()
+        {
+            const int NumberOfFiles = 5;
+            const int NumberOfRows = 3;
+            const int ExpectedRowCount = NumberOfFiles * NumberOfRows;
+
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString + "poolingEnabled=false";
+                conn.Open();
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Arrange
+                    var data = string.Concat(Enumerable.Repeat(string.Join(",", "TestData") + "\n", NumberOfRows));
+                    var prefix = $"{Path.GetTempPath()}{Guid.NewGuid()}";
+                    for (int i = 0; i < NumberOfFiles; i++)
+                    {
+                        File.WriteAllText(prefix + $"_{i}.csv", data);
+                    }
+                    CreateOrReplaceTable(conn, TableName, new[] { "COL1 STRING" });
+                    cmd.CommandText = $"PUT file://{prefix + "_*.csv"} @%{TableName} AUTO_COMPRESS=FALSE";
+                    var reader = cmd.ExecuteReader();
+
+                    // Act
+                    cmd.CommandText = $"COPY INTO {TableName} FROM @%{TableName} PATTERN='.*.csv' FILE_FORMAT=(TYPE=CSV)";
+                    int actualRowCount = cmd.ExecuteNonQuery();
+
+                    // Assert
+                    Assert.AreEqual(ExpectedRowCount, actualRowCount);
+
+                    for (int i = 0; i < NumberOfRows; i++)
+                    {
+                        File.Delete(prefix + $"_{i}.csv");
+                    }
```

Comment:
we should remove the files from the server as well.

If you would put the files into a subfolder with a name with a random part we could just remove this folder.

Reply:
Done

Reply:
I don't see the change, maybe you forgot to push it?